src/error.rs

```diff
@@ -17,15 +17,59 @@ pub enum Rav1dError {
     /// which is more optimal since `0` is no error for [`Dav1dResult`].
     EGeneric = 1,
 
-    ENOENT = libc::ENOENT as u8,
-    EIO = libc::EIO as u8,
-    EAGAIN = libc::EAGAIN as u8,
-    ENOMEM = libc::ENOMEM as u8,
-    EINVAL = libc::EINVAL as u8,
-    ERANGE = libc::ERANGE as u8,
-    ENOPROTOOPT = libc::ENOPROTOOPT as u8,
+    /// Invalid buffer.
+    ///
+    /// No Sequence Header OBUs were found in the buffer.
+    InvalidBuffer = libc::ENOENT as u8,
+
+    /// Can't open file.
+    ///
+    /// IO error.
+    CantOpenFile = libc::EIO as u8,
+    /// Try again.
+    ///
+    /// If this is returned by [`Decoder::send_data`] or [`Decoder::send_pending_data`] then there
+    /// are decoded frames pending that first have to be retrieved via [`Decoder::get_picture`]
+    /// before processing any further pending data.
+    ///
+    /// If this is returned by [`Decoder::get_picture`] then no decoded frames are pending
+    /// currently and more data needs to be sent to the decoder.
+    Again = libc::EAGAIN as u8,
+    /// Not enough memory.
+    ///
+    /// Not enough memory is currently available for performing this operation.
+    NotEnoughMemory = libc::ENOMEM as u8,
+    /// Invalid argument.
+    ///
+    /// One of the arguments passed to the function was invalid.
+    InvalidArgument = libc::EINVAL as u8,
+    /// Out of range.
+    ///
+    /// Frame size is larger than the limit
+    OutOfRange = libc::ERANGE as u8,
+    /// Unsupported bitstream.
+    ///
+    /// The provided bitstream is not supported by `rav1d`.
+    UnsupportedBitstream = libc::ENOPROTOOPT as u8,
 }
 
+impl std::fmt::Display for Rav1dError {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Rav1dError::Again => write!(fmt, "Try again"),
+            Rav1dError::InvalidArgument => write!(fmt, "Invalid argument"),
+            Rav1dError::NotEnoughMemory => write!(fmt, "Not enough memory available"),
+            Rav1dError::UnsupportedBitstream => write!(fmt, "Unsupported bitstream"),
+            Rav1dError::EGeneric => write!(fmt, "Generic error"),
+            Rav1dError::InvalidBuffer => write!(fmt, "Invalid buffer"),
+            Rav1dError::CantOpenFile => write!(fmt, "Can't open file: IO error"),
+            Rav1dError::OutOfRange => write!(fmt, "Out of range"),
```

Comment:
Could you pull these out into a `pub const fn as_str(&self) -> &'static str` method?